chain/types/ethtypes/eth_1559_transactions_test.go

```diff
@@ -0,0 +1,241 @@
+	}
+}
+
+func TestTransformParams(t *testing.T) {
+	constructorParams, err := actors.SerializeParams(&evm.ConstructorParams{
+		Initcode: mustDecodeHex("0x1122334455"),
+	})
+	require.Nil(t, err)
+
+	evmActorCid, ok := actors.GetActorCodeID(actorstypes.Version10, "reward")
+	require.True(t, ok)
+
+	params, err := actors.SerializeParams(&init10.ExecParams{
+		CodeCID:           evmActorCid,
+		ConstructorParams: constructorParams,
+	})
+	require.Nil(t, err)
+
+	var exec init10.ExecParams
+	reader := bytes.NewReader(params)
+	err1 := exec.UnmarshalCBOR(reader)
+	require.Nil(t, err1)
+
+	var evmParams evm.ConstructorParams
+	reader1 := bytes.NewReader(exec.ConstructorParams)
+	err1 = evmParams.UnmarshalCBOR(reader1)
+	require.Nil(t, err1)
+
+	require.Equal(t, mustDecodeHex("0x1122334455"), evmParams.Initcode)
+}
+
+func TestEcRecover(t *testing.T) {
+	rHex := "0x479ff7fa64cf8bf641eb81635d1e8a698530d2f219951d234539e6d074819529"
+	sHex := "0x4b6146d27be50cdbb2853ba9a42f207af8d730272f1ebe9c9a78aeef1d6aa924"
+	fromHex := "0x3947D223fc5415f43ea099866AB62B1d4D33814D"
+	v := byte(0)
+
+	msgHex := "0x02f1030185012a05f2008504a817c800825208942b87d1cb599bc2a606db9a0169fcec96af04ad3a880de0b6b3a764000080c0"
+	pubKeyHex := "0x048362749392a0e192eff600d21155236c5a0648d300a8e0e44d8617712c7c96384c75825dc5c7595df2a5005fd8a0f7c809119fb9ab36403ed712244fc329348e"
+
+	msg := mustDecodeHex(msgHex)
+	pubKey := mustDecodeHex(pubKeyHex)
+	r := mustDecodeHex(rHex)
+	s := mustDecodeHex(sHex)
+	from := mustDecodeHex(fromHex)
+
+	sig := append(r, s...)
+	sig = append(sig, v)
+	require.Equal(t, 65, len(sig))
+
+	sha := sha3.NewLegacyKeccak256()
+	sha.Write(msg)
+	h := sha.Sum(nil)
+
+	pubk, err := gocrypto.EcRecover(h, sig)
+	require.Nil(t, err)
+	require.Equal(t, pubKey, pubk)
+
+	sha.Reset()
+	sha.Write(pubk[1:])
+	h = sha.Sum(nil)
+	h = h[len(h)-20:]
+
+	require.Equal(t, from, h)
+}
+
+func TestDelegatedSigner(t *testing.T) {
+	rHex := "0xcf1fa52fae9154ba21d67aeca9b42adfe186eb9e426c441051a8473efd190848"
+	sHex := "0x0e6c8c79ffaf35fb8f136c8cf6c5656f1f3befad21f2644321aa6dba58d68737"
+	v := byte(0)
+
+	msgHex := "0x02f08401df5e76038502540be400843b9aca008398968094ff000000000000000000000000000000000003f2832dc6c080c0"
+	pubKeyHex := "0x04cfecc0520d906cbfea387759246e89d85e2998843e56ad1c41de247ce10b3e4c453aa73c8de13c178d94461b6fa3f8b6f74406ce43d2fbab6992d0b283394242"
+
+	msg := mustDecodeHex(msgHex)
+	pubk := mustDecodeHex(pubKeyHex)
+	r := mustDecodeHex(rHex)
+	s := mustDecodeHex(sHex)
+
+	addrHash, err := EthAddressFromPubKey(pubk)
+	require.NoError(t, err)
+
+	from, err := address.NewDelegatedAddress(builtintypes.EthereumAddressManagerActorID, addrHash)
+	require.NoError(t, err)
+
+	sig := append(r, s...)
+	sig = append(sig, v)
+	require.Equal(t, 65, len(sig))
+
+	signature := &crypto1.Signature{
+		Type: crypto1.SigTypeDelegated,
+		Data: sig,
+	}
+
+	err = sigs.Verify(signature, from, msg)
+	require.NoError(t, err)
+}
+
+func prepareTxTestcases() ([]TxTestcase, error) {
```

Comment:
any idea where these came from and if it would be easy to compile a similar list for legacy?